editoast/src/views/search.rs

```diff
@@ -692,5 +743,73 @@ pub(super) struct SearchResultItemScenario {
     object(name = "project", config = SearchResultItemProject),
     object(name = "study", config = SearchResultItemStudy),
     object(name = "scenario", config = SearchResultItemScenario),
+    object(name = "trainschedule", config = SearchResultItemTrainSchedule),
 )]
 pub struct SearchConfigFinder;
+
+#[cfg(test)]
+pub mod test {
+
+    use axum::http::StatusCode;
+    use pretty_assertions::assert_eq;
+    use rstest::rstest;
+    use serde_json::json;
+
+    use super::*;
+    use crate::models::fixtures::{create_simple_train_schedule, create_timetable};
+    use crate::models::train_schedule::TrainSchedule;
+    use crate::views::test_app::TestAppBuilder;
+
+    #[rstest]
+    async fn search_trainschedule_post() {
+        let app = TestAppBuilder::default_app();
+        let pool = app.db_pool();
+
+        // Create the timetable in the database
+        let timetable = create_timetable(&mut pool.get_ok()).await;
+        let timetable_id = timetable.id;
+
+        // Load data of the json file
+        #[derive(serde::Deserialize)]
+        struct TrainScheduleInfo {
+            pub train_name: String,
+        }
+        let train_schedule_info: TrainScheduleInfo =
+            serde_json::from_str(include_str!("../tests/train_schedules/simple.json"))
+                .expect("Unable to parse");
+        let train_name = train_schedule_info.train_name;
+
+        // Add a train_schedule in the database
+        let train_schedule: TrainSchedule =
+            create_simple_train_schedule(&mut pool.get_ok(), timetable_id).await;
+        assert_eq!(train_schedule.train_name, train_name);
+
+        // The body
+        let request = app.post("/search").json(&json!({
+            "object": "trainschedule",
+            "query": [
+                "and",
+                [
+                    "search",
+                    [
+                        "train_name"
+                    ],
+                    train_name
+                ],
+                [
+                    "=",
+                    [
+
+                        "timetable_id"
+                    ],
+                    timetable_id
+                ]
+            ],
+        }));
```

Comment:
```suggestion
        let request = app.post("/search").json(&json!({
            "object": "trainschedule",
            "query": ["and", ["=", ["train_name"], train_name],
                             ["=", ["timetable_id"], timetable_id]],
        }));
```

If rustfmt allows it, it's probably more readable like so. Also the `search` operator is only meant to be used with columns indexed to allow fuzzy matching.